### DISCUSSION TITLE
Gatsby itself changes the url on the page after it is loaded

### DISCUSSION THREAD

**Discussion by Frontend-Aleksandr-G** [gatsbyjs/gatsby#38802] [Help]
My page in gatsby node

for (const browser of browsers) {
  for (const locale of locales) {
  actions.createPage({
    path: `/${browser}/${locale}`,
    component: path.resolve('./src/templates/Main.tsx'),
    context: {
      browser,
      locale,
      messages: await getMessages(locale),
      page: 'home',
    },
  })
On the server there is a logic that uses the request headers to give the necessary page and on the path / gives the necessary Main.tsx template the problem is that after loading the page path changes for example to /chrome/en/ with / How to disable this

**Comment by 808vita** [comment]
So you want to remove trailing slash , right ?
From gatsby doc [trailingSlash](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-config/#trailingslash).
Hope this helps.


```
gatsby-config.js

module.exports = {
 trailingSlash: "never",
}
```

trailingSlash
Support added in gatsby@4.7.0

Configures the creation of URLs for pages, and whether to remove, append, or ignore trailing slashes.

always: Always add trailing slashes to each URL, e.g. /x to /x/.
never: Remove all trailing slashes on each URL, e.g. /x/ to /x.
ignore: Don’t automatically modify the URL
The default setting for this option is always. Gatsby Cloud automatically handles and supports the trailingSlash option. Alternate hosting providers (or if you’re managing this on your own) should follow the “Redirects, and expected behaviour from the hosting provider”.

**Reply by Frontend-Aleksandr-G** [reply]
No i need stop auto replace url after load page. On browser i get url /  but after loading page url change on /chrome/en how stop this
